CMakeLists.txt

```diff
@@ -591,15 +593,38 @@ if(WITH_OPTIM)
             endif()
         endif()
     elseif(BASEARCH_PPC_FOUND)
+        # Common arch detection code
+        if(WITH_ALTIVEC)
+            check_ppc_intrinsics()
+        endif()
         if(WITH_POWER8)
             check_power8_intrinsics()
+        endif()
+        if(HAVE_VMX OR HAVE_POWER8_INTRIN)
+            list(APPEND ZLIB_ARCH_HDRS ${ARCHDIR}/power.h)
+            list(APPEND ZLIB_ARCH_SRCS ${ARCHDIR}/power.c)
+        endif()
+        # VMX specific options and files
+        if(WITH_ALTIVEC)
+            if(HAVE_VMX)
+                add_definitions(-DPPC_FEATURES)
+                if(HAVE_ALTIVEC)
+                    add_definitions(-DPPC_VMX_ADLER32)
+                    add_definitions(-DPPC_VMX_SLIDEHASH)
+                    set(PPC_SRCS ${ARCHDIR}/adler32_vmx.c ${ARCHDIR}/slide_vmx.c)
+                    list(APPEND ZLIB_ARCH_SRCS ${PPC_SRCS})
+                    add_feature_info(ALTIVEC 1 "Support the AltiVec instruction set, using \"-maltivec\"")
+                    set_property(SOURCE ${PPC_SRCS} PROPERTY COMPILE_FLAGS "${PPCFLAGS}")
+                endif()
```

Comment:
Should have:
```
else()
     set(WITH_ALTIVEC OFF)
endif()
```
If `NOT HAVE_ALTIVEC`.

Reply:
Also it may be missing `feature_info` at the bottom of CMakeLists.txt.

Reply:
If I set `WITH_ALTIVEC` to `OFF`, I should be able to use that as second parameter of `add_feature_info` and move that line outside `if` block.